### ISSUE TITLE
Add machine validation check for unsigned software executing

### ISSUE THREAD

**Issue by hpaul-osi** [osisoft/PI-Security-Audit-Tools#249]
[Lee Holmes' tweet](https://twitter.com/Lee_Holmes/status/875781096885043201) points out that PowerShell can easily audit whether unsigned software is executing on the system.

```
gps | gi -ea ig | Get-AuthenticodeSignature | Group {$_.SignerCertificate.Subject } | Select Count,Name
```

This could be a separate machine validation check, or an enhancement to the application whitelisting check (AU10004).

**Comment by jdryden-osi** [comment]
It looks like Get-AuthenticodeSignature will report many Microsoft OS executables as unsigned, at least on Windows versions before Win10/Server2016 from my quick testing. This will require more research to avoid false positives.

**Comment by hpaul-osi** [comment]
You're correct, the [AppX packages](https://stackoverflow.com/questions/38634052/what-is-the-structure-of-appxsignature-p7x) and [packages authenticated through security catalog files](https://blog.didierstevens.com/2008/01/11/the-case-of-the-missing-digital-signatures-tab/) present a ~~problem~~ challenge here.  Get-AuthenticodeSignature only supports embedded signatures at present.  There is a [PowerShell UserVoice](https://blogs.msdn.microsoft.com/windowsmobile/2006/05/17/programmatically-checking-the-authenticode-signature-on-a-file/) item to address catalog signatures, but does not have a lot of votes, so we can't count on it getting implemented any time soon.   

One possibility is using C# via Add-Type if Constrained Language Mode is not in use.  This may actually be a good trade off since Constrained Language Mode should only be activated if they have an Application Whitelisting policy defined, in which case they are less likely to permit execution of unsigned code anyway.

**Comment by hpaul-osi** [comment]
Get-AuthenticodeSignature supports catalog files in PSv5. AppX is still not supported for PSv3, PSv4 or PSv5, but these apps are in a predictable location %programfiles%\WindowsApps, so for the purpose of this check, we can ignore anything in that location. Similarly, in PSv3/PSv4 we can ignore anything in %windir%\system32. The main goal of this check is to highlight unsigned (unverified) applications executing, not to identify malware or tampered files.
